primitives/src/assets.rs

```diff
@@ -424,17 +399,32 @@ where
     }
 
     #[inline]
-    fn can_withdraw(
+    fn can_reduce_by_amount(
         asset_id: AssetId,
         account: &C::AccountId,
         amount: Balance,
+        existence_requirement: ExistenceRequirement,
     ) -> Result<(), FungibleLedgerError> {
         Self::ensure_valid(asset_id)?;
-        FungibleLedgerError::from_withdraw(if asset_id == A::NativeAssetId::get() {
-            <Native as FungibleInspect<C::AccountId>>::can_withdraw(account, amount)
+
+        let keep_alive = match existence_requirement {
+            ExistenceRequirement::KeepAlive => true,
+            ExistenceRequirement::AllowDeath => false,
+        };
+        let reducible_amount = if asset_id == A::NativeAssetId::get() {
```

Comment:
This means available amount from users, right? Doesn't include like locked or reserved tokens.

Reply:
Locked or reserved tokens cannot be withdrawn so they are not part of the reducible amount,